glyphs-reader/plist_derive/src/lib.rs

```diff
@@ -2,157 +2,102 @@ extern crate proc_macro;
 
 use proc_macro2::TokenStream;
 use quote::{quote, quote_spanned};
-use syn::spanned::Spanned;
-use syn::{parse_macro_input, Data, DeriveInput, Fields};
+use std::iter;
+use syn::{parse_macro_input, spanned::Spanned, Data, DeriveInput, Fields};
 
 mod attrs;
 
 #[proc_macro_derive(FromPlist, attributes(fromplist))]
-pub fn derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+pub fn from_plist(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
 
-    let deser = match add_deser(&input) {
+    let field_cases = match add_fieldcases(&input) {
         Ok(thing) => thing,
         Err(e) => return e.into_compile_error().into(),
     };
 
     let name = input.ident;
 
     let expanded = quote! {
-        impl crate::from_plist::FromPlist for #name {
-            fn from_plist(plist: crate::plist::Plist) -> Self {
-                let mut map = plist.expect_dict().unwrap();
-                #name {
-                    #deser
+        impl FromPlist for #name {
+            fn parse(tokenizer: &mut Tokenizer<'_>) -> Result<Self, crate::plist::Error> {
+                use crate::plist::Error;
+
+                if !tokenizer.consume(b'{') {
```

Comment:
although I agree it is not worth the change at this point, I do think that having direct parsing code in all of these impls is suboptimal.

Reply:
The bulk of it is struct.field = parse value; I'm not sure how to avoid that. We could migrate the surrounding boilerplate out I suppose. Leaving it alone for now.